packages/render-fidelity-tools/src/artifact-creator.ts

```diff
@@ -48,168 +48,186 @@ export class ArtifactCreator {
+
+      await this.compareRenderers(scenario);
 
-      let golden;
       try {
-        golden = await fs.readFile(goldenPath);
+        const autoTestResult = await this.autoTest(scenario);
+        autoTestResults.results.push(autoTestResult);
       } catch (error) {
-        throw new Error(`❌ Failed to read ${rendererName}'s ${
-            scenarioName} golden! Error message: ${error.message}`);
+        const message = `Fail to analyze scenario :${
+            scenarioName}! Error message: ${error.message}`;
+
+        if (error.message === WARNING_MESSAGE) {
+          autoTestResults.warnings.push(message);
+        } else {
+          autoTestResults.errors.push(message);
+        }
       }
 
-      const screenshotImage = pngjs.PNG.sync.read(screenshot).data;
-      const goldenImage = pngjs.PNG.sync.read(golden).data;
+      analyzedScenarios.push(scenario);
+    }
 
-      const imageDimensions = {
-        width: dimensions.width * DEVICE_PIXEL_RATIO,
-        height: dimensions.height * DEVICE_PIXEL_RATIO
-      };
-      const comparator =
-          new ImageComparator(screenshotImage, goldenImage, imageDimensions);
+    console.log('💾 Recording configuration');
 
-      await fs.writeFile(
-          join(outputDirectory, scenarioName, goldenConfig.file), golden);
+    const finalConfig: ImageComparisonConfig =
+        Object.assign({}, this.config, {scenarios: analyzedScenarios});
 
-      const {analysis} = comparator.analyze();
-      const {rmsDistanceRatio} = analysis;
-      console.log(
-          `\n  📊 Decibels of root mean square color distance (without threshold): ${
-              (10 * Math.log10(rmsDistanceRatio)).toFixed(2)}`);
+    await fs.writeFile(
+        join(outputDirectory, 'config.json'), JSON.stringify(finalConfig));
 
-      analysisResults.push(analysis);
-    }
+    await fs.writeFile(
+        join(outputDirectory, 'autoTestResults.json'),
+        JSON.stringify(autoTestResults));
+  }
+
+  protected async analyze(
+      candidateImage: Buffer, goldenImage: Buffer, dimensions: Dimensions,
+      semiTransparentCheck: boolean = true): Promise<ImageComparisonAnalysis> {
+    const imageDimensions = {
+      width: dimensions.width * DEVICE_PIXEL_RATIO,
+      height: dimensions.height * DEVICE_PIXEL_RATIO
+    };
+    const comparator =
+        new ImageComparator(candidateImage, goldenImage, imageDimensions);
+
+    const {analysis} = comparator.analyze(semiTransparentCheck);
+    const {rmsDistanceRatio} = analysis;
+    console.log(
+        `\n  📊 Decibels of root mean square color distance (without threshold): ${
+            (10 * Math.log10(rmsDistanceRatio)).toFixed(2)}`);
 
-    return analysisResults;
+    return analysis;
   }
 
   async captureScreenshot(
       renderer: string, scenarioName: string, dimensions: Dimensions,
-      outputPath: string = join(this.outputDirectory, 'model-viewer.png'),
+      outputPath: string|null = join(this.outputDirectory, 'model-viewer.png'),
```

Comment:
just dont want to write the model-viewer's screenshot into result folder anymore because it's not used in the dev page